### PR TITLE
2146-Hardcode-class-rewriter-in-RB-class

### LINKED ISSUE #2146: RB classes should not hardcode class rewriter and searcher....
We should reduce the number of references to `RBParseTreeRewriter` and `RBParseTreeSearcher`

### PR DESCRIPTION
**Pull request by LaurineDargaud**
fix: #2146
From 62 to 23 references to RBParseTreeRewriter **with good name**: #parseTreeRewriterClass

### CODE REVIEW

**Comment by Ducasse** [comment]
This is strange since it did not build.

**Comment by Ducasse** [comment]
rekicking the build. There is something strange.

File: src/Refactoring-Core/RBAddParameterRefactoring.class.st [hunk 1]
```diff
@@ -128,7 +128,7 @@ RBAddParameterRefactoring >> newSelectorString [
 { #category : #private }
 RBAddParameterRefactoring >> parseTreeRewriter [
 	| rewriteRule oldString newString |
-	rewriteRule := RBParseTreeRewriter new.
+	rewriteRule := self parseTreeRewriter.
```
Comment (guillep) [inline src/Refactoring-Core/RBAddParameterRefactoring.class.st]: This looks like an infinite loop ;)

File: src/Refactoring-Core/RBChangeMethodNameRefactoring.class.st [hunk 2]
```diff
@@ -62,7 +62,7 @@ RBChangeMethodNameRefactoring >> newSelector [
 { #category : #private }
 RBChangeMethodNameRefactoring >> parseTreeRewriter [
 	| rewriteRule oldString newString |
-	rewriteRule := RBParseTreeRewriter new.
+	rewriteRule := self parseTreeRewriter.
```
Comment (guillep) [inline src/Refactoring-Core/RBChangeMethodNameRefactoring.class.st]: This one too

File: src/Refactoring-Core/RBRenameMethodRefactoring.class.st [hunk 3]
```diff
@@ -63,8 +63,8 @@ RBRenameMethodRefactoring >> parseTreeRewriter [
 	newString := self buildSelectorString: newSelector
 				withPermuteMap: permutation.
 	rewriteRule := self hasPermutedArguments 
-				ifTrue: [RBParseTreeRewriter new]
-				ifFalse: [RBParseTreeRewriter replaceLiteral: oldSelector with: newSelector].
+				ifTrue: [self parseTreeRewriter]
```
Comment (guillep) [inline src/Refactoring-Core/RBRenameMethodRefactoring.class.st]: and this one :)

**Review by guillep — CHANGES_REQUESTED** [review]
Some infinite loops over here and there :)

File: src/Refactoring-Core/RBAddParameterRefactoring.class.st (hunk 1 above)
Comment (LaurineDargaud) [reply src/Refactoring-Core/RBAddParameterRefactoring.class.st]: OK

File: src/Refactoring-Core/RBChangeMethodNameRefactoring.class.st (hunk 2 above)
Comment (LaurineDargaud) [reply src/Refactoring-Core/RBChangeMethodNameRefactoring.class.st]: OK

File: src/Refactoring-Core/RBRenameMethodRefactoring.class.st (hunk 3 above)
Comment (LaurineDargaud) [reply src/Refactoring-Core/RBRenameMethodRefactoring.class.st]: OK

**Review by guillep — APPROVED** [review]
Ok!

**Comment by guillep** [comment]
Tests are fixed! There is a fleaky test only. I'll open an issue for that.

**Pull request merged**